tests/replication2.erl

```diff
@@ -11,592 +11,692 @@
+  ?assertEqual(0, repl_util:wait_for_reads(BFirst, 201, 300, TestBucket, 2)),
+
+  %% Get the leader for Cluster "B"
+  LeaderB = rpc:call(BFirst, riak_core_cluster_mgr, get_leader, []),
+
+  log_to_nodes(ANodes ++ BNodes, "Testing client failover: stopping ~p", [LeaderB]),
+
+  lager:info("Testing client failover: stopping ~p", [LeaderB]),
+  rt:stop(LeaderB),
+  rt:wait_until_unpingable(LeaderB),
+  BSecond = hd(BNodes -- [LeaderB]),
+  repl_util:wait_until_leader(BSecond),
+
+  LeaderB2 = rpc:call(BSecond, riak_core_cluster_mgr, get_leader, []),
+
+  lager:info("New leader is ~p", [LeaderB2]),
+  ?assertEqual(ok, repl_util:wait_until_connection(LeaderA2)),
+
+  lager:info("Writing 100 more keys to ~p now that the old leader is down", [ASecond]),
+  ?assertEqual([], repl_util:do_write(ASecond, 301, 400, TestBucket, 2)),
+
+  %% Verify data is replicated to Cluster B
+  lager:info("Reading 101 keys written to ~p from ~p", [ASecond, BSecond]),
+  ?assertEqual(0, repl_util:wait_for_reads(BSecond, 301, 400, TestBucket, 2)),
+
+  log_to_nodes(ANodes ++ BNodes, "Test fullsync with ~p and ~p down", [LeaderA, LeaderB]),
+  lager:info("Re-running fullsync with ~p and ~p down", [LeaderA, LeaderB]),
+
+  repl_util:start_and_wait_until_fullsync_complete(LeaderA2),
+
+  %% This says test full sync, but there's never a verification step.
+  log_to_nodes(ANodes ++ BNodes, "Test fullsync after restarting ~p", [LeaderA]),
+
+  %% Put everything back to 'normal'.
+  lager:info("Nodes restarted"),
+  lager:info("Restarting down nodes ~p, ~p", [LeaderA, LeaderB]),
+  rt:start(LeaderA),
+  rt:start(LeaderB),
+  rt:wait_until_pingable(LeaderA),
+  rt:wait_until_pingable(LeaderB),
+  rt:wait_for_service(LeaderA, [riak_kv, riak_repl]),
+  repl_util:start_and_wait_until_fullsync_complete(LeaderA2).
+
+
+%% @doc Network Partition Test
+%%      Test Cycle:
+%%        Connect the Cluster "A" leader and swap its cookie for a new cookie.
+%%        Disconnect node with new cookie from the rest of the nodes.
+%%        Nodes will elect a new leader.
+%%        Reset cookie on disconnected node and reconnct.
+%%        Write 2 keys to node that was reconnected.
+%%        Verify replication of keys to Cluster "B".
+network_partition_test( [AFirst|_] = ANodes, [BFirst|_] = BNodes ) ->
+
+  TestHash =  list_to_binary([io_lib:format("~2.16.0b", [X]) || <<X>> <= erlang:md5(term_to_binary(os:timestamp()))]),
+  TestBucket = <<TestHash/binary, "-network_partition_test">>,
+  LeaderA = rpc:call(AFirst, riak_core_cluster_mgr, get_leader, []),
+
+  log_to_nodes(ANodes ++ BNodes, "Starting network partition test"),
+
+  %% @todo add stuff
+  %% At this point, realtime sync should still work, but, it doesn't because of a bug in 1.2.1
+  %% Check that repl leader is LeaderA
+  %% Check that LeaderA2 has ceeded socket back to LeaderA
+
+  %%lager:info("Leader: ~p", [rpc:call(ASecond, riak_core_cluster_mgr, get_leader, [])]),
```

Comment:
This comment block was in the original module. I assumed that it had some special significance at one point.  It has no meaning to me but I was erring on the side of caution, in case it was referring to something that I am ignorant about.